pkgs/servers/web-apps/matomo/default.nix

```diff
@@ -0,0 +1,59 @@
+{ stdenv, fetchurl, makeWrapper, php }:
+
+stdenv.mkDerivation rec {
+  name = "matomo-${version}";
+  version = "3.3.0";
+
+  src = fetchurl {
+    # TODO: As soon as the tarballs are renamed as well on future releases, this should be enabled again
+    # url = "https://builds.matomo.org/${name}.tar.gz";
+    url = "https://builds.matomo.org/piwik-${version}.tar.gz";
+    sha256 = "1ybzj3kk0x29nv8c6xnhir5d9dr0q0fl1vnm4i7zvhml73ryqk0f";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  # make-localhost-default-database-server.patch:
+  #   This changes the default value of the database server field
+  #   from 127.0.0.1 to localhost.
+  #   unix socket authentication only works with localhost,
+  #   but password-based SQL authentication works with both.
+  # TODO: is upstream interested in this?
```

Comment:
Maybe they're more sceptical, though, as 127.0.0.1 could be considered more failsafe or something. :laughing: This requires that `localhost` resolves properly to `::1` and `127.0.0.1` and such. (It's not too long ago when NixOS itself had no such entries in `/etc/hosts` as well…) Also this could be bad if MySQL is not listening on IPv6. I'll eventually try what they think, though, the PR would be simple enough. :slightly_smiling_face:

Reply:
Maybe a compile time option, preserving the behavior of 127..1, where --unix-auth or something expects a hosts mapping and changes it to localhost. Although that's a little more complicated :)